storage/url/url.go

```diff
@@ -277,6 +284,9 @@ func (u *URL) MarshalJSON() ([]byte, error) {
 
 // HasGlob checks if a string contains any wildcard chars.
 func (u *URL) HasGlob() bool {
+	if u.origin != nil {
```

Comment:
we can revert this change. callers must call url.Origin().HasGlob() if the intention is to check source origin has glob.

Reply:
The intention here is to recursively call origin URL. Because fs behaves differently from the s3 client. We both clone the original URL after expanding glob([ref](https://github.com/peak/s5cmd/blob/65fa203c5b053060563f99b9cf2588da25633030/storage/fs.go#L89)) and walking directory ([ref](https://github.com/peak/s5cmd/blob/65fa203c5b053060563f99b9cf2588da25633030/storage/fs.go#L113)). That is why source URL has an origin and also origin can keep its origin :D.   When I check just the origin without this logic, some of the tests are failing.

Reply:
done. How about a explicit SetOrigin() method similar to SetRelative() ? Can you take a look?